signer/client/signer_trust.go

```diff
@@ -189,3 +194,46 @@ func (trust *NotarySigner) CheckHealth(timeout time.Duration) error {
 
 	return err
 }
+
+// CheckHealth are used to probe whether the server is able to handle rpcs.
+func (trust *NotarySigner) CheckHealth(d time.Duration, serviceName string) error {
+
+	// The client can optionally set the service name it wants to query for health status
+	if serviceName != "" {
+		out, err := healthCheck(d, trust.healthClient, serviceName)
```

Comment:
can we define constants for `grpc.health.v1.Health.KeyManagement` and `grpc.health.v1.Health.Signer`, and then switch on `serviceName` to see if matches either one of those values?

We can define `healthCheckSigner` and `healthCheckKeyManager` to specifically check the health of those services since we only have those two, and then error out if the client provides an invalid `serviceName`